packages/flutter_tools/lib/src/protocol_discovery.dart

```diff
@@ -81,15 +89,21 @@ class ProtocolDiscovery {
   ///
   /// When a new observatory URL: is available in [logReader],
   /// the URLs are forwarded at most once every [throttleDuration].
+  /// Returns when no event has been observed for [throttleTimeout].
   ///
   /// Port forwarding is only attempted when this is invoked,
   /// for each observatory URL in the stream.
   Stream<Uri> get uris {
-    return _uriStreamController.stream
+    Stream<Uri> uriStream = _uriStreamController.stream
```

Comment:
@blasten does this change look okay to you?  On iOS we don't want the logs to scan forever when no observatory URL is found since it needs to detect the URL wasn't found.